lib/facter/yum_updates.rb

```diff
@@ -0,0 +1,37 @@
+Facter.add('yum_package_updates') do
+  confine osfamily: 'RedHat'
+  setcode do
+    yum_updates = []
+
+    if File.executable?('/usr/bin/yum')
+      yum_get_result = Facter::Util::Resolution.exec('/usr/bin/yum -y -q -C list updates 2>&1')
```

Comment:
Why do you need stderr here mixing into stdout?

Reply:
The redirect seems useless, I removed it.

I also switched to long arguments as advised.